policybot/handlers/githubwebhook/filters/resulttester/resultTester.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resulttester
+
+import (
+	"context"
+
+	webhook "github.com/go-playground/webhooks/github"
+
+	"istio.io/bots/policybot/handlers/githubwebhook/filters"
+	"istio.io/bots/policybot/pkg/config"
+	"istio.io/bots/policybot/pkg/gh"
+	"istio.io/bots/policybot/pkg/storage"
+	"istio.io/bots/policybot/pkg/storage/cache"
+	"istio.io/bots/policybot/pkg/testresults"
+	"istio.io/pkg/log"
+)
+
+// Updates the DB based on incoming GitHub webhook events.
+type ResultTester struct {
+	store      storage.Store
+	repos      map[string]bool
+	cache      *cache.Cache
+	ght        *gh.ThrottledClient
+	ctx        context.Context
+	bucketName string
```

Comment:
This field needs to be in a different struct, otherwise it makes this component such that it will not behave properly if invoked N times in parallel. Look in pkg/syncer/syncer.go and you'll see the syncState struct which is created for this purpose.

Basically, you want to explicitly separate the stable state from the per-request state so that you can process multiple requests in parallel.